src/app/check-in/check-in.component.ts

```diff
@@ -13,96 +13,86 @@ import { ThsDataService } from '../services/ths-data.service';
+  public badEmail: boolean = false;
+  
+  constructor(private router: Router, private tsDataService: TsScreenerDataService, private tfiDataService: TfiDataService, private thsDataService: ThsDataService) {};
+  
+  public onClick() {
+    if(this.isCredentialsValid()) {
+      sessionStorage.clear();
+      this.tsDataService.clearHistory();
+      this.tfiDataService.clearHistory();
+      this.thsDataService.clearHistory();
+      Utilities.setSessionStorage('patient-id', this.patientId);
+      Utilities.setSessionStorage('first-name', this.firstName);
+      Utilities.setSessionStorage('last-name', this.lastName);
+      Utilities.setSessionStorage('email', this.email);
+      this.router.navigateByUrl('/appointments');
+    } else {
+      this.authenticationFlag = false;
+      this.patientId = '';
+    }
+  }
 
-      if (thd < 5) {
-        thd += 4;
-      } else { thd -= 2; }
+  isCredentialsValid() : boolean {
+    this.badPatientId = false;
+    this.badEmail = false;
 
-      if (snd < 7) {
-        snd += 2;
-      } else { snd -= 3; }
+    let result = true;
 
-      if (fst < 7) {
-        fst += 2;
-      } else { fst -= 2; }
+    // Check these individually so we can set the corresponding flags
+    // and still return the result.
+    if(!this.isPatientIdValid()) {
+      this.badPatientId = true;
+      result = false;
+    }
 
-      let result1 = thd.toString();
-      let result2 = fst.toString();
-      let result3 = frt.toString();
-      let result4 = snd.toString();
+    if(!this.isEmailValid()) {
+      this.badEmail = true;
+      result = false;
+    }
 
-      return result1 + result2 + result3 + result4;
-   }
-   public idGenerator() {
-     let hash = 0;
-     let first = this.firstName;
-     let last = this.lastName;
-     // name = this.firstName.charCodeAt(0) + this.lastName.charCodeAt(0);
-     if (this.firstName == null || this.lastName == null || this.lastFourSS == null) {
-       return hash;
-     }
-     // for (let i = 0; i < nameLength; i++) {
-     //   hash += Math.pow(name.charCodeAt(i) * 31, nameLength - i);
-     // }
-     let result = first.substr(0, 3) + last.substr(0, 3) + this.numRearrange(this.lastFourSS);
-     return result;
-   }
+    return result;
+  }
 
-   /**
-    * This function will be call when the "check in" button is pressed.
-    * It evaluates the value is typed in the input area; if the value has the length of 6,
-    * then the page will be route to the "audiologist" page;
-    * if the value of length 4, then the page will be route to the "appointment" page.
-    * If the value has different length, then it will be informed as incorrect check in.
-    */
+  isEmailValid() : boolean {
+    let numAtSymbols : number = 0;
 
-   public onClick() {
+    // Email is an optional field the user does not need
+    // to fill out. However, should they choose to do so, it
+    // should follow a proper format.
+    if(this.email.length === 0)
+      return true;
 
```

Comment:
My suggestion would be to declare email to be optional 
`class CheckInComponent {
public email?: string;
}`
but, it isn't a big deal.